pkg/executables/cmk.go

```diff
@@ -119,39 +135,147 @@ func (c *Cmk) ValidateAffinityGroupsPresent(ctx context.Context, domain string,
+	}
+	if multipleZone {
+		filterArgs = append(filterArgs, fmt.Sprintf("type=\"%s\"", "Shared"))
+	}
+	// account must be specified within a domainId
+	// domainId can be specified without account
+	if len(domainId) > 0 {
+		filterArgs = append(filterArgs, fmt.Sprintf("domainid=\"%s\"", domainId))
+		if len(account) > 0 {
+			filterArgs = append(filterArgs, fmt.Sprintf("account=\"%s\"", account))
+		}
+	}
+	zoneId := ""
+	if zoneRef.Zone.Type == v1alpha1.Id {
+		zoneId = zoneRef.Zone.Value
 	} else {
-		filterArg = fmt.Sprintf("name=\"%s\"", zone.Value)
+		for _, zoneIdentifier := range zones {
+			if zoneRef.Zone.Value == zoneIdentifier.Name {
+				zoneId = zoneIdentifier.Id
+				break
+			}
+		}
 	}
-	result, err := c.exec(ctx, "list", "zones", filterArg)
+	filterArgs = append(filterArgs, fmt.Sprintf("zoneid=\"%s\"", zoneId))
+	result, err := c.exec(ctx, filterArgs...)
 	if err != nil {
-		return fmt.Errorf("error getting zones info: %v", err)
+		return fmt.Errorf("error getting network info: %v", err)
 	}
 	if result.Len() == 0 {
-		return fmt.Errorf("zone %s not found", zone)
+		if multipleZone {
+			return fmt.Errorf("%s network %s not found in zoneRef %s", "Shared", zoneRef.Network.Value, zoneRef.Zone.Value)
+		} else {
+			return fmt.Errorf("network %s not found in zoneRef %s", zoneRef.Network.Value, zoneRef.Zone.Value)
+		}
 	}
 
 	response := struct {
-		CmkZones []cmkZone `json:"zone"`
+		CmkNetworks []cmkNetwork `json:"network"`
 	}{}
 	if err = json.Unmarshal(result.Bytes(), &response); err != nil {
 		return fmt.Errorf("failed to parse response into json: %v", err)
 	}
-	zones := response.CmkZones
-	if len(zones) > 1 {
-		return fmt.Errorf("duplicate zone %s found", zone)
-	} else if len(zones) == 0 {
-		return fmt.Errorf("zone %s not found", zone)
+	networks := response.CmkNetworks
+
+	// filter by network name -- cmk does not support name= filter
+	if zoneRef.Network.Type == v1alpha1.Name {
+		networks = []cmkNetwork{}
+		for _, net := range response.CmkNetworks {
+			if net.Name == zoneRef.Network.Value {
+				networks = append(networks, net)
+			}
+		}
+	}
+
+	if len(networks) > 1 {
+		return fmt.Errorf("duplicate network %s found", zoneRef.Network.Value)
+	} else if len(networks) == 0 {
+		if multipleZone {
+			return fmt.Errorf("%s network %s not found in zoneRef %s", "Shared", zoneRef.Network.Value, zoneRef.Zone.Value)
+		} else {
+			return fmt.Errorf("network %s not found in zoneRef %s", zoneRef.Network.Value, zoneRef.Zone.Value)
+		}
 	}
 	return nil
 }
 
-// TODO: Add support for domain filtering
-func (c *Cmk) ValidateAccountPresent(ctx context.Context, account string) error {
-	result, err := c.exec(ctx, "list", "accounts", fmt.Sprintf("name=\"%s\"", account))
+func (c *Cmk) ValidateAccountPresent(ctx context.Context, account string, domainId string) error {
+	result, err := c.exec(ctx, "list", "accounts", fmt.Sprintf("name=\"%s\"", account), fmt.Sprintf("domainid=\"%s\"", domainId))
```

Comment:
what happens if domain isn't set? Does that break this function/cmk? Or is it required in the cluster spec?